src/Microsoft.Identity.Web/MicrosoftIdentityCircuitHandler.cs

```diff
@@ -17,257 +17,257 @@
 
-            if (microsoftIdentityWebChallengeUserException == null)
-            {
+			if (microsoftIdentityWebChallengeUserException == null)
+			{
 #pragma warning disable CA1062 // Validate arguments of public methods
-                microsoftIdentityWebChallengeUserException = exception.InnerException as MicrosoftIdentityWebChallengeUserException;
+				microsoftIdentityWebChallengeUserException = exception.InnerException as MicrosoftIdentityWebChallengeUserException;
 #pragma warning restore CA1062 // Validate arguments of public methods
-            }
+			}
 
-            if (microsoftIdentityWebChallengeUserException != null &&
-               IncrementalConsentAndConditionalAccessHelper.CanBeSolvedByReSignInOfUser(microsoftIdentityWebChallengeUserException.MsalUiRequiredException))
-            {
-                var properties = IncrementalConsentAndConditionalAccessHelper.BuildAuthenticationProperties(
-                    microsoftIdentityWebChallengeUserException.Scopes,
-                    microsoftIdentityWebChallengeUserException.MsalUiRequiredException,
-                    User,
-                    microsoftIdentityWebChallengeUserException.Userflow);
+			if (microsoftIdentityWebChallengeUserException != null &&
+			   IncrementalConsentAndConditionalAccessHelper.CanBeSolvedByReSignInOfUser(microsoftIdentityWebChallengeUserException.MsalUiRequiredException))
+			{
+				var properties = IncrementalConsentAndConditionalAccessHelper.BuildAuthenticationProperties(
+					microsoftIdentityWebChallengeUserException.Scopes,
+					microsoftIdentityWebChallengeUserException.MsalUiRequiredException,
+					User,
+					microsoftIdentityWebChallengeUserException.Userflow);
 
-                List<string> scopes = properties.Parameters.ContainsKey(Constants.Scope) ? (List<string>)properties.Parameters[Constants.Scope]! : new List<string>();
-                string claims = properties.Parameters.ContainsKey(Constants.Claims) ? (string)properties.Parameters[Constants.Claims]! : string.Empty;
-                string userflow = properties.Items.ContainsKey(OidcConstants.PolicyKey) ? properties.Items[OidcConstants.PolicyKey]! : string.Empty;
+				List<string> scopes = properties.Parameters.ContainsKey(Constants.Scope) ? (List<string>)properties.Parameters[Constants.Scope]! : new List<string>();
+				string claims = properties.Parameters.ContainsKey(Constants.Claims) ? (string)properties.Parameters[Constants.Claims]! : string.Empty;
+				string userflow = properties.Items.ContainsKey(OidcConstants.PolicyKey) ? properties.Items[OidcConstants.PolicyKey]! : string.Empty;
 
-                ChallengeUser(
-                    scopes.ToArray(),
-                    claims,
-                    userflow);
-            }
-            else
-            {
-                throw exception;
-            }
-        }
+				ChallengeUser(
+					scopes.ToArray(),
+					claims,
+					userflow);
+			}
+			else
+			{
+				throw exception;
```

Comment:
This line is a problem because it causes the original stacktrace to be lost. The exception needs to be thrown with `throw;` instead of `throw exception;` which means it can't be done inside this method and instead needs to be done from a catch block directly.

Reply:
we are given the exception to process here for incremental consent, not sure how we can move this to a try/catch. otherwise, developers would have to do this, and it becomes too complicated.